src/TheSpace/TheSpace.sol

```diff
@@ -112,15 +114,58 @@ contract TheSpace is HarbergerMarket {
             return new uint256[](0);
         }
         uint256 left = total - offset;
-        uint256 pageSize = left > limit ? limit : left;
+        uint256 size = left > limit ? limit : left;
 
-        uint256[] memory tokens = new uint256[](pageSize);
+        uint256[] memory tokens = new uint256[](size);
 
-        for (uint256 i = 0; i < pageSize; i++) {
+        for (uint256 i = 0; i < size; i++) {
             uint256 tokenIndex = i + offset;
             tokens[i] = tokenOfOwnerByIndex(owner, tokenIndex);
         }
 
         return tokens;
     }
+
+    /**
+     * @notice Get owned pixels for a user using pagination.
+     *
+     * @dev use getTokensByOwner internally
+     * @return total owned pixel total amount for this user
+     * @return size page size
+     * @return page page number
+     * @return pixels pixels in this page
+     */
+    function getPixelsPageByOwner(
+        address owner_,
+        uint256 size_,
+        uint256 page_
+    )
+        external
+        view
+        returns (
+            uint256 total,
+            uint256 size,
+            uint256 page,
+            Pixel[] memory pixels
+        )
+    {
+        if (page_ == 0) {
+            return (balanceOf(owner_), size_, page_, new Pixel[](0));
+        }
+
+        uint256[] memory _tokens = getTokensByOwner(owner_, size_, size_ * (page_ - 1));
+        Pixel[] memory _pixels = new Pixel[](_tokens.length);
+        for (uint256 i = 0; i < _tokens.length; i++) {
```

Comment:
`getTokensByOwner` already did a for loop, I think we can move [L112-L119](https://github.com/thematters/contracts/blob/74456f0778384fc435a14d6e0d38f36ce7046a52/src/TheSpace/TheSpace.sol#L112-L119=) to a internal function and reused by `getTokensByOwner` and `getPixelsPageByOwner`. 

Or just copy to `getPixelsPageByOwner`:

```solidity
        if (page_ == 0) {
            return (balanceOf(owner_), size_, page_, new Pixel[](0));
        }

        uint256 total = balanceOf(owner);
        if (offset >= total) {
            return new uint256[](0);
        }
        uint256 left = total - offset;
        uint256 size = left > limit ? limit : left;

        uint256[] memory _tokens = getTokensByOwner(owner_, size_, size_ * (page_ - 1));
        Pixel[] memory _pixels = new Pixel[](_tokens.length);

        for (uint256 i = 0; i < size; i++) {
            uint256 tokenIndex = i + offset;
            uint256 tokenId = tokenOfOwnerByIndex(owner, tokenIndex);
            tokens[i] = tokenId;
            _pixels[i] = _getPixel(tokenId);
        }

        return (total, size_, page_, _pixels);
```